tox.ini

```diff
@@ -14,7 +14,8 @@ extras =
     testing
     coverage: coverage
 deps =
-    pipprevious: pip==22.2.*
+    pipsupported: pip==24.2
+    piplowest: pip==22.2.*
```

Comment:
What does "supported" actually mean here?

`pyproject.toml` requires `pip >= 22.2`, so that maps neatly onto `piplowest`, but the this lowest is also supported.

Is "pipsupported" the highest version we claim to support?

Like, there's a pip 24.3, but we don't claim to support that (yet)?

If so, would `piphighest` be clearer, and mirror `piplowest` better?

(and with dashes for readability)

Reply:
> What does "supported" actually mean here?
> 
> `pyproject.toml` requires `pip >= 22.2`, so that maps neatly onto `piplowest`, but the this lowest is also supported.
> 
> Is "pipsupported" the highest version we claim to support?
> 
> Like, there's a pip 24.3, but we don't claim to support that (yet)?
>

Yes, kind of. I just wanted to have something that is pinned in CI. So after this effort, when a PR fails the CI, it's clear that it's not related to something that said PR changed. I don't want to think too much about this. The immediate goal is to unblock development, and it can be refined later.

> If so, would `piphighest` be clearer, and mirror `piplowest` better?

Maybe, I was considering it. But in order to avoid spending a lot of time overthinking, I've gone with "supported" as "something we run in CI".

> (and with dashes for readability)

I was reminded recently that tox makes dash-separated chunks factors, which might not be desired. I thought about it promptly and decided that it's not worthy of being in the scope of this PR and blocking it. But it could be debated separately, once this is complete. I'm currently blocked on getting #2106 to a working state (plus have some work to do in `pypi-publish`, which is mostly what's distracting me from this PR).

Reply:
Thanks, sounds good. Let's check this again when the other things are in place 👍